services/shhext/service_test.go

```diff
@@ -449,3 +451,91 @@ func waitForHashInTracker(track *tracker, hash common.Hash, state EnvelopeState,
 		}
 	}
 }
+
+func TestRequestMessagesSync(t *testing.T) {
+	suite.Run(t, new(RequestMessagesSyncSuite))
+}
+
+type RequestMessagesSyncSuite struct {
+	suite.Suite
+
+	localAPI  *PublicAPI
+	localNode *enode.Node
+	remoteRW  *p2p.MsgPipeRW
+}
+
+func (s *RequestMessagesSyncSuite) SetupTest() {
+	db, err := leveldb.Open(storage.NewMemStorage(), nil)
+	s.Require().NoError(err)
+	conf := &whisper.Config{
+		MinimumAcceptedPOW: 0,
+		MaxMessageSize:     100 << 10,
+	}
+	w := whisper.New(conf)
+	pkey, err := crypto.GenerateKey()
+	s.Require().NoError(err)
+	node := enode.NewV4(&pkey.PublicKey, net.ParseIP("127.0.0.1"), 1, 1)
+	peer := p2p.NewPeer(node.ID(), "1", []p2p.Cap{{"shh", 6}})
+	rw1, rw2 := p2p.MsgPipe()
+	errorc := make(chan error, 1)
+	go func() {
+		err := w.HandlePeer(peer, rw2)
+		errorc <- err
+	}()
+	// 0 - statusCode
+	s.Require().NoError(p2p.ExpectMsg(rw1, 0, []interface{}{whisper.ProtocolVersion, math.Float64bits(w.MinPow()), w.BloomFilter(), false, true}))
+	s.Require().NoError(p2p.SendItems(rw1, 0, whisper.ProtocolVersion, whisper.ProtocolVersion, math.Float64bits(w.MinPow()), w.BloomFilter(), true, true))
+
+	service := New(w, nil, db, params.ShhextConfig{})
+
+	s.localAPI = NewPublicAPI(service)
+	s.localNode = node
+	s.remoteRW = rw1
+}
+
+func (s *RequestMessagesSyncSuite) TestExpired() {
+	// intentionally discarding all requests, so that request will timeout
+	go func() {
+		msg, err := s.remoteRW.ReadMsg()
+		s.Require().NoError(err)
+		msg.Discard()
+	}()
+	s.Require().EqualError(s.localAPI.RequestMessagesSync(RetryConf{
+		BaseTimeout: time.Second,
+	}, MessagesRequest{
+		MailServerPeer: s.localNode.String(),
+	}), "failed to request messages after 1 retries")
```

Comment:
Just curious, shouldn't this error message be something like `unexpectedly managed to retrieve messages`?

Reply:
why? this one is the opposite, method failed to request messages and notifies caller about that